LV_analysis/1_get_multiplier_LV_coverage_Pa.py

```diff
@@ -0,0 +1,335 @@
+# (compare two blue boxes between plots)
+generic_nonzero = all_coverage_df[all_coverage_df["gene type"] == "generic"][
+    "nonzero LV coverage"
+].values
+generic_highweight = all_coverage_df[all_coverage_df["gene type"] == "generic"][
+    "highweight LV coverage"
+].values
+
+(stats, pvalue) = scipy.stats.ttest_ind(generic_nonzero, generic_highweight)
+print(pvalue)
+
+# +
+# Test: mean number of LVs generic genes present in vs mean number of LVs other genes high weight in
+# (compare blue and grey boxes in high weight plot)
+other_highweight = all_coverage_df[all_coverage_df["gene type"] == "other"][
+    "highweight LV coverage"
+].values
+generic_highweight = all_coverage_df[all_coverage_df["gene type"] == "generic"][
+    "highweight LV coverage"
+].values
+
+(stats, pvalue) = scipy.stats.ttest_ind(other_highweight, generic_highweight)
+print(pvalue)
+
+# +
+# Check that coverage of other and generic genes across all LVs is NOT signficantly different
+# (compare blue and grey boxes in nonzero weight plot)
+other_nonzero = all_coverage_df[all_coverage_df["gene type"] == "other"][
+    "nonzero LV coverage"
+].values
+generic_nonzero = all_coverage_df[all_coverage_df["gene type"] == "generic"][
+    "nonzero LV coverage"
+].values
+
+(stats, pvalue) = scipy.stats.ttest_ind(other_nonzero, generic_nonzero)
+print(pvalue)
+# -
+
+# ## Get LVs that generic genes are highly weighted in
+#
+# Since we are using quantiles to get high weight genes per LV, each LV has the same number of high weight genes. For each set of high weight genes, we will get the proportion of generic vs other genes. We will select the LVs that have a high proportion of generic genes to examine.
+
+# Get proportion of generic genes per LV
+prop_highweight_generic_dict = lv.get_prop_highweight_generic_genes(
+    processed_dict_genes, multiplier_model_z, quantile_threshold
+)
+
+# Return selected rows from summary matrix
+multiplier_model_summary = pd.read_csv(
+    "multiplier_Pa_model_summary.tsv", sep="\t", index_col=0, header=0
+)
+lv.create_LV_df(
+    prop_highweight_generic_dict,
+    multiplier_model_summary,
+    0.5,
+    "Generic_LV_summary_table_Pa.tsv",
+)
+
+# Plot distribution of weights for these nodes
+node = "LV30"
+lv.plot_dist_weights(
+    node,
+    multiplier_model_z,
+    shared_genes,
+    20,
+    all_coverage_df,
+    f"weight_dist_{node}.svg",
+)
+
+# ## Save
+
+# +
+# Save plot
+nonzero_fig.figure.savefig(
+    nonzero_figure_filename,
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+
+# Save plot
+highweight_fig.figure.savefig(
+    highweight_figure_filename,
+    format="svg",
+    bbox_inches="tight",
+    transparent=True,
+    pad_inches=0,
+    dpi=300,
+)
+# -
+
+# **Takeaway:**
+# * In the first nonzero boxplot, generic and other genes are present in a similar number of LVs. This isn't surprising since the number of genes that contribute to each LV is <1000.
```

Comment:
This isn't what the statistical test shows, is it (see previous comment)? I guess if you don't believe the results of the t-test that's valid, the medians on the box plot don't look terribly different at least by eye, but it might be good to say that explicitly (unless I'm mistaken about what you're claiming here).

Reply:
Yes I will clarify this comment. You're correct that despite the t-test saying there is a significant difference, the distribution looks very similar. So I wasn't planning to claim a difference despite the t-test statistic.

Reply:
Yeah - I think the sample size is fairly high (~10,000 genes, I'm guessing?) so even a small difference will be significant. That doesn't necessarily invalidate the test, but it seems valid to argue that the difference is small enough that it is unlikely to matter biologically.

[This stackexchange thread](https://stats.stackexchange.com/a/2519) might be interesting, along those lines.

Reply:
Thank you for sharing this. This is very interesting. I think I'll need to think about this some more, its not really intuitive to me why increasing sample size might lead to increased chance of significance even for a small effect. I guess my understanding is that the t-test is comparing means between two distributions, so I'm not sure why increasing sample size might yield this